CalculatorPresenter.cs

```diff
@@ -186,45 +159,48 @@ private void OnSaveHistoryClick(object sender, EventArgs e)
         private void OnMemoryClick(object sender, EventArgs e)
         {
             Button b = (Button)sender;
-            string memoryClick = b.Text;
+            WhenMemoryClick(b.Text);
+        }
+
+
+        public void WhenMemoryClick(string memoryClick)
+        {
             string mesaj = "Doresti sa golesti memoria?";
             string titlu = "Golire Memorie";
 
-            if (!_calculatorModel.IsMemoryStored)
+            if (!_isMemoryStored)
             {
                 _calculatorView.DisableMemoryButtons();
             }
 
             switch (memoryClick)
             {
                 case "MC":
-                    _calculatorModel.IsMemoryStored = _messageBoxDisplayService.PromptUser(mesaj, titlu);
+                    _businessLogicObject.MemoryClear();
+                    _isMemoryStored = _messageBoxDisplayService.PromptUser(mesaj, titlu);
                     _calculatorView.DisableMemoryButtons();
                     break;
                 case "MR":
-                    _textAfisat = _utils.FormatShownText(_calculatorModel.Memory);
+                    _textAfisat = _businessLogicObject.MemoryRestore();
                     _calculatorView.SetResultBoxText(_textAfisat);
                     break;
                 case "MS":
-                    _calculatorModel.Memory = (decimal)Double.Parse(_textAfisat);
-                    _calculatorModel.IsMemoryStored = true;
+                    _businessLogicObject.MemoryStore(_textAfisat);
+                    _isMemoryStored = true;
```

Comment:
Intai de toate ca as tine memoria in calculator, e treaba lui.

Apoi ca as pune *dupa* switch o linie de genul `if (_businessLogicObject.HasMemoryStored()) <enable> else <disable>`. Asa nu mai e treaba presenter-ului sa tina minte daca are sau nu memorie stocata acest calculator.